query-engine/core/src/query_graph_builder/read/utils.rs

```diff
@@ -37,6 +37,31 @@ pub(crate) fn collect_selected_fields(
     }
 }
 
+/// Creates a `FieldSelection` from a query selection, which contains only scalar fields.
+/// Automatically adds model IDs to the selected fields as well.
+/// Unwraps are safe due to query validation.
```

Comment:
That's good to know, but it doesn't look like we're calling `.unwrap()` anywhere here, am I missing anything?